python/doc/examples/reliability_sensitivity/flood_model.ipynb

```diff
@@ -0,0 +1,300 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Estimate a probability with the flood model\n",
+    "\n",
+    "In this example, we estimate the probability that the ouput of a function exceeds a given threshold with the FORM method. We consider the flooding model."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Introduction\n",
+    "\n",
+    "The following figure presents the dyke that protects industrial facilities. When the river height exceeds the one of the dyke, flooding occurs. The model is based on a crude simplification of the 1D hydrodynamical equations of Saint-Venant under the assumptions of uniform and constant flowrate and large rectangular sections.\n",
+    "\n",
+    "<img src=\"river_section_adjusted_light.png\" width=\"400\" />\n",
+    "\n",
+    "Four independent random variables are considered:\n",
+    "\n",
+    " - Q (Flow rate) [m^3 s^-1]\n",
+    " - Ks (Strickler) [m^1/3 s^-1]\n",
+    " - Zv (downstream height) # [m]\n",
+    " - Zm (upstream height) # [m]\n",
+    "\n",
+    "When the Strickler coefficient increases, the riverbed generates less friction. \n",
+    "\n",
+    "We consider the four following parameters:\n",
+    "\n",
+    "* the height of the dyke : Hd = 3 [m],\n",
+    "* the level of the banks of the river Zb = 55.5 [m],\n",
+    "* the length of the river L = 5000 [m],\n",
+    "* the width of the river B = 300 [m].\n",
+    "\n",
+    "The height of the dyke is:\n",
+    "\n",
+    "$$\n",
+    "Z_d = Z_b + H_d\n",
+    "$$\n",
+    "\n",
+    "The slope of the river is assumed to be close to zero, which implies:\n",
+    "\n",
+    "$$\n",
+    "\\alpha = \\frac{Z_m - Z_v}{L},\n",
+    "$$\n",
+    "\n",
+    "if $Z_m \\geq Z_v$. \n",
+    "\n",
+    "The height of the water level is:\n",
+    "\n",
+    "$$\n",
+    "H = \\left(\\frac{Q}{K_s B \\sqrt{\\alpha}}\\right)^{0.6},\n",
+    "$$\n",
+    "\n",
+    "for any $K_s, Q>0$.\n",
+    "\n",
+    "Therefore, the height of the flood is:\n",
+    "\n",
+    "$$\n",
+    "Z_c = H + Z_v\n",
+    "$$\n",
+    "\n",
+    "The water level is greater than the height of the dyke (i.e. there is a flood) if:\n",
+    "\n",
+    "$$\n",
+    "S = Z_c - Z_d\n",
+    "$$\n",
+    "\n",
+    "is greater than zero.\n",
+    "\n",
+    "If we substitute the parameters into the equation, we get:\n",
+    "\n",
+    "$$\n",
+    "S = \\left(\\frac{Q}{300 Ks \\sqrt{(Zm-Zv)/5000}}\\right)^{3/5} +Zv-55.5-3.\n",
```

Comment:
Why are Z_b and H_b substituted but not any other variables? This seems arbitrary.

Reply:
The parameters L=5000 and B=300 are substituted too. The parameters L, B, Hd and Zb are substituted because these are not random variables. Another possibility would be to define a function with 8 inputs and then create a parametric function where L, B Hd and Zb are set as parameters. This would be (a little) more complicated.